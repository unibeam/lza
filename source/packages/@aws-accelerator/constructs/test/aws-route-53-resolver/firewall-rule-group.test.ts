/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import {
  ResolverFirewallRuleGroup,
  ResolverFirewallRuleGroupAssociation,
} from '../../lib/aws-route-53-resolver/firewall-rule-group';
import { snapShotTest } from '../snapshot-test';

const testNamePrefix = 'Construct(ResolverFirewallRuleGroup): ';

const ruleProps = {
  action: 'BLOCK',
  firewallDomainListId: 'TestDomainList',
  priority: 101,
  blockResponse: 'NXDOMAIN',
};

const stack = new cdk.Stack();

const ruleGroup = new ResolverFirewallRuleGroup(stack, 'TestRuleGroup', {
  firewallRules: [ruleProps],
  name: 'TestRuleGroup',
  tags: [],
});

new ResolverFirewallRuleGroupAssociation(stack, 'TestRuleGroupAssoc', {
  firewallRuleGroupId: ruleGroup.groupId,
  priority: 101,
  vpcId: 'TestVpc',
});

describe('ResolverFirewallRuleGroup', () => {
  snapShotTest(testNamePrefix, stack);
});
