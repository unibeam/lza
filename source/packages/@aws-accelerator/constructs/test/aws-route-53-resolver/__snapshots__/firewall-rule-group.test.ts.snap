// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResolverFirewallRuleGroup Construct(ResolverFirewallRuleGroup):  Snapshot Test 1`] = `
{
  "Resources": {
    "TestRuleGroup43F9213A": {
      "Properties": {
        "FirewallRules": [
          {
            "Action": "BLOCK",
            "BlockResponse": "NXDOMAIN",
            "FirewallDomainListId": "TestDomainList",
            "Priority": 101,
          },
        ],
        "Tags": [
          {
            "Key": "Name",
            "Value": "TestRuleGroup",
          },
        ],
      },
      "Type": "AWS::Route53Resolver::FirewallRuleGroup",
    },
    "TestRuleGroupAssoc48F4678D": {
      "Properties": {
        "FirewallRuleGroupId": {
          "Ref": "TestRuleGroup43F9213A",
        },
        "Priority": 101,
        "VpcId": "TestVpc",
      },
      "Type": "AWS::Route53Resolver::FirewallRuleGroupAssociation",
    },
  },
}
`;
